File: backend/app/models/score.py
from pydantic import BaseModel
from typing import Literal
from datetime import datetime
import uuid


Module = Literal['quiz', 'fake_or_real', 'spot_zone']


class SubmitScorePayload(BaseModel):
    username: str
    score: int
    total: int
    module: Module


class Score(BaseModel):
    id: str
    username: str
    score: int
    total: int
    module: Module
    timestamp: str

    @classmethod
    def from_payload(cls, payload: SubmitScorePayload) -> 'Score':
        return cls(
            id=str(uuid.uuid4()),
            username=payload.username,
            score=payload.score,
            total=payload.total,
            module=payload.module,
            timestamp=datetime.utcnow().isoformat(),
        )

File: application/src/types/score.ts
export type Module = 'quiz' | 'fake_or_real' | 'spot_zone'

export interface Score {
  id: string
  username: string
  score: number
  total: number
  module: Module
  timestamp: string
}

export interface SubmitScorePayload {
  username: string
  score: number
  total: number
  module: Module
}

export interface Stats {
  total_games: number
  average_score: number
  hardest_question_id: number | null
}
